src/luadebug/rdebug_hookmgr.cpp

```diff
@@ -463,17 +464,19 @@ struct hookmgr {
         default:
             return;
         }
-        push_callback(L);
-        luadebug::debughost::set(L, hL);
```

Comment:
Here should always call back into lua. If not, then it shouldn't have gotten here.

Reply:
You are right.
Rechecking the code it seems like there's a bug in luajit's implementation, in specific on how returning from C call is being handled, have to find a better solution